File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/auth/SshPrivateKeyScmAuth.java
package com.tencent.devops.scm.api.pojo.auth;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SshPrivateKeyScmAuth implements IScmAuth {
    public static final String CLASS_TYPE = "SSH_PRIVATEKEY";

    private String privateKey;
    private String passphrase;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/auth/SshPrivateKeyScmAuth.kt
package com.tencent.devops.scm.api.pojo.auth

class SshPrivateKeyScmAuth(
    val privateKey: String,
    val passphrase: String? = null
) : IScmAuth {
    companion object {
        const val CLASS_TYPE = "SSH_PRIVATEKEY"
    }
}
